packages/contracts-bedrock/scripts/New.sol

```diff
@@ -0,0 +1,220 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import { Vm } from "forge-std/Vm.sol";
+import { ResourceMetering } from "src/L1/ResourceMetering.sol";
+import "src/libraries/DisputeTypes.sol";
+
+library New {
+    Vm internal constant vm = Vm(0x7109709ECfa91a80626fF3989D68f67F5b1DD12D);
+
+    error CreateFailed(string);
+    error Create2Failed(string);
+
+    function _create(bytes memory _code) internal returns (address addr_) {
+        assembly {
+            addr_ := create(0, add(_code, 0x20), mload(_code))
+        }
+        require(addr_ != address(0), "New: cannot create");
+    }
+
+    function _create2(bytes memory _code, bytes32 _salt) internal returns (address addr_) {
+        assembly {
+            addr_ := create2(0, add(_code, 0x20), mload(_code), _salt)
+        }
+    }
+
+    function addressManager() internal returns (address addr_) {
+        bytes memory code = vm.getCode("AddressManager.sol:AddressManager");
+        addr_ = _create({ _code: code });
+        if (addr_ == address(0)) revert CreateFailed("AddressManager");
+    }
+
+    function proxyAdmin(address _owner) internal returns (address addr_) {
+        bytes memory code = vm.getCode("ProxyAdmin.sol:ProxyAdmin");
+        bytes memory args = abi.encode(_owner);
+        code = bytes.concat(code, args);
+        addr_ = _create({ _code: code });
+        if (addr_ == address(0)) revert CreateFailed("ProxyAdmin");
+    }
```

Comment:
_I know this is a draft but came across this so leaving one comment now_

I don't love this idea because we lose type safety of constructor args now. In theory tests should catch that, but before merging this PR we should:
1. Post before/after benchmarks of build times with `default` and `lite` profiles to ensure the tradeoff is worth it
2. Make sure traces are still decoded properly. Forge fuzzy matches bytecode from artifacts when decoding, and since the artifacts we load are from the currently-running profile tracing should work the same, but worth sanity checking

Reply:
Thanks for the quick review here. I don't love this either but it is an exploration on how to reduce compile time. Now the bytecode for each of the contracts should not be inlined in `CommonTest`. This didn't seem to speed up the compilation time to what it was previously but need to check out an older commit to measure. The next thing I would like to try is not have `CommonTest` inherit from `Deploy` and instead deploy `Deploy`. That could help to reduce the size of `CommonTest` and hopefully the compile time, since so many tests inherit from `CommonTest`  that means that there are many very large contracts

Reply:
>  I don't love this either but it is an exploration on how to reduce compile time.

Totally understand, did not mean to shoot down the idea! Otherwise all sounds great and agreed on the idea/plan here :)